src/test/java/com/googleinterns/zoomtube/servlets/TranscriptServletTest.java

```diff
@@ -0,0 +1,254 @@
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+@RunWith(JUnit4.class)
+public final class TranscriptServletTest {
+  @Rule public final MockitoRule mockito = MockitoJUnit.rule();
+
+  @Mock private HttpServletRequest request;
+  @Mock private HttpServletResponse response;
+
+  private TranscriptServlet servlet;
+  private DatastoreService datastore;
+  private Gson gson;
+  private StringWriter lectureTranscript;
+
+  private static final LocalDatastoreServiceTestConfig datastoreConfig =
+      (new LocalDatastoreServiceTestConfig()).setNoStorage(true);
+  private static final LocalServiceTestHelper localServiceHelper =
+      new LocalServiceTestHelper(datastoreConfig);
+  private static final String LECTURE_ID_A = "123";
+  private static final String LECTURE_ID_B = "345";
+  private static final String LECTURE_ID_C = "234";
+  private static final String SHORT_VIDEO_ID = "Obgnr9pc820";
+  private static final String LONG_VIDEO_ID = "jNQXAC9IVRw";
+  private static final String SHORT_VIDEO_JSON =
+      "[{\"transcriptKey\":{\"kind\":\"TranscriptLine\",\"id\":1},"
+      + "\"lectureKey\":{\"kind\":\"lecture\",\"id\":123},"
+      + "\"start\":\"0.4\",\"duration\":\"1\",\"content\":\" \"},"
+      + "{\"transcriptKey\":{\"kind\":\"TranscriptLine\",\"id\":2},\"lectureKey\":"
+      + "{\"kind\":\"lecture\",\"id\":123},\"start\":\"2.28\",\"duration\":\"1\",\"content\""
+      + ":\"Hi\"},{\"transcriptKey\":{\"kind\":\"TranscriptLine\",\"id\":3},\"lectureKey\":"
+      + "{\"kind\":\"lecture\",\"id\":123},\"start\":\"5.04\",\"duration\":\"1.6\","
+      + "\"content\":\"Okay\"}]";
+  private static final String LONG_VIDEO_JSON =
+      "[{\"transcriptKey\":{\"kind\":\"TranscriptLine\",\"id\":1},\"lectureKey\":"
+      + "{\"kind\":\"lecture\",\"id\":123},\"start\":\"1.3\",\"duration\":"
+      + "\"3.1\",\"content\":\"All right, so here we are\\nin front of the "
+      + "elephants,\"},{\"transcriptKey\":{\"kind\":\"TranscriptLine\",\"id\":4},\"lectureKey\":"
+      + "{\"kind\":\"lecture\",\"id\":123},\"start\":\"12.7\",\"duration\":\"4.3\",\"content\":"
+      + "\"and that&#39;s, that&#39;s cool.\"},{\"transcriptKey\":{\"kind\":\"TranscriptLine\",\"id\":5},"
+      + "\"lectureKey\":{\"kind\":\"lecture\",\"id\":123},\"start\":\"17\",\"duration\":\"1.767\",\""
+      + "content\":\"And that&#39;s pretty much all there is to say.\"},{\"transcriptKey\":{\"kind\""
+      + ":\"TranscriptLine\",\"id\":2},\"lectureKey\":{\"kind\":\"lecture\",\"id\":123},\"start\":"
+      + "\"4.4\",\"duration\":\"4.766\",\"content\":\"the cool thing about these guys\\nis that "
+      + "they have really,\"},{\"transcriptKey\":{\"kind\":\"TranscriptLine\",\"id\":3},\"lectureKey\":"
+      + "{\"kind\":\"lecture\",\"id\":123},\"start\":\"9.166\",\"duration\":\"3.534\","
+      + "\"content\":\"really, really long trunks,\"}]";
```

Comment:
I would add a TODO to represent this differently. I worry if anyone wants to change this data in the future it would be difficult. Maybe creating helper methods for creating this strings is the way to go. Not sure.

Reply:
Can you have a helper method for creating these objects and the helper method takes as input the variables you want to test?

Reply:
+1 I agree, that is why a TODO is good in this case.